### PR TITLE
jobs/build: parallelize some stages

### PR DESCRIPTION
**Pull request by dustymabe**
- QEMU and QEMU upgrade tests
- keylime hashlist generation and building the qemu image
- Uploading to Clouds (AWS/GCP)
- Kicking off Cloud Provider test jobs

### CODE REVIEW

**Comment by dustymabe** [comment]
currently untested.

File: jobs/build.Jenkinsfile [hunk 1]
```diff
@@ -305,36 +322,44 @@ lock(resource: "build-${params.STREAM}") {
...
         }
 
-        stage('Kola:QEMU') {
-            // leave 512M for overhead; VMs are 1G each
-            def parallel = ((cosa_memory_request_mb - 512) / 1024) as Integer
+        // reset for the next batch of independent tasks
+        parallelruns = [:]
+
+        // Kola QEMU tests
+        parallelruns['Kola:QEMU'] = {
```
Comment (jlebon) [inline jobs/build.Jenkinsfile:329]: We should really look at moving this over to `fcosKola` which already knows to parallelize with upgrade tests.

**Review by jlebon — APPROVED** [review]
I didn't review the syntax in details, but the high-level looks sane to me!

File: jobs/build.Jenkinsfile (hunk 1 above)
Comment (dustymabe) [reply jobs/build.Jenkinsfile:329]: yeah. I'm torn here. I like that all of the actual commands that we know and love are in this file without a level of indirection, but there is a lot of value in the library too.

**Pull request merged**